src/cli.rs

```diff
@@ -293,24 +325,70 @@ pub struct Cli {
+    ///
+    /// The URL scheme defaults to "http://" normally, or "https://" if
+    /// the program is invoked as "xhs".
+    ///
+    /// A leading colon works as shorthand for localhost. ":8000" is equivalent
+    /// to "localhost:8000", and ":/path" is equivalent to "localhost/path".
+    ///
+    /// URLs can have a leading "://" which allows quickly converting a URL
+    /// into a valid xh or HTTPie command. For example "http://httpbin.org/json"
+    /// becomes "http ://httpbin.org/json".
     #[clap(value_name = "[METHOD] URL")]
     raw_method_or_url: String,
 
     /// Optional key-value pairs to be included in the request
     ///
-    ///   - key==value to add a parameter to the URL
-    ///   - key=value to add a JSON field (--json) or form field (--form)
-    ///   - key:=value to add a complex JSON value (e.g. `numbers:=[1,2,3]`)
-    ///   - key@filename to upload a file from filename (with --form)
-    ///   - @filename to use a file as the request body
-    ///   - header:value to add a header
-    ///   - header: to unset a header
-    ///   - header; to add a header with an empty value
+    /// The separator is used to determine the type i.e. header, request body, query string, etc.
+    ///
+    /// Possible REQUEST_ITEM types are:
+    ///
+    ///     key==value
+    ///         Add a query string to the URL.
+    ///
+    ///     key=value
+    ///         Add a JSON property (--json) or form field (--form) to
+    ///         the request body.
+    ///
+    ///     key=@file
+    ///         Add a JSON property (--json) or form field (--form) from a
+    ///         file to the request body.
     ///
-    /// A backslash can be used to escape special characters (e.g. weird\:key=value).
+    ///     key:=value
+    ///         Add a field with literal JSON value to the request body,
+    ///         e.g. numbers:=[1,2,3] enabled:=true.
+    ///
+    ///     key:=@file
+    ///         Add a field with literal JSON value from a file to the
+    ///         request body.
+    ///
+    ///     key@file
+    ///         Upload a file (requires --form or --multipart).
+    ///
+    ///         To set the filename and mimetype, ";type=" and
+    ///         ";filename=" can be used respectively, e.g
+    ///         pfp@ra.jpg;type=image/jpeg;filename=profile.jpg
+    ///
+    ///     @filename
+    ///         Use a file as the request body.
+    ///
+    ///     header:value
+    ///         Add a header, e.g. user-agent:foobar
+    ///
+    ///     header:
+    ///         Unset a header, e.g. connection:
+    ///
+    ///     header;
+    ///         Add a header with an empty value.
+    ///
+    /// A backslash can be used to escape special characters, e.g. weird\:key=value.
+    ///
+    /// To construct a complex JSON object, the REQUEST_ITEM's key can be set to a JSON path instead of a field name.
+    ///
+    /// For more information on the nested json syntax, refer to https://httpie.io/docs/cli/nested-json.
```

Comment:
These two paragraphs have much more to do with each other than with the paragraph right before, so I think it's clearest to merge them:
```suggestion
    /// To construct a complex JSON object, the REQUEST_ITEM's key can be set to a JSON path instead of a field name.
    /// For more information on this syntax, refer to https://httpie.io/docs/cli/nested-json.
```